frappe/www/app.py

```diff
@@ -27,8 +27,7 @@ def get_context(context):
 	try:
 		boot = frappe.sessions.get()
 	except Exception as e:
-		boot = frappe._dict(status="failed", error=str(e))
-		print(frappe.get_traceback())
+		raise frappe.SessionBootFailed from e
```

Comment:
This weird re-raising is required! 

Otherwise permission/404 errors wont give proper error on the user side.